infrastructure/monitoring/kube-prometheus-stack/README.md

```diff
@@ -0,0 +1,264 @@
+In the following, we will setup Alertmanager, Grafana and Prometheus to use Keycloak as identity provider for the authentication. Grafana can natively use Keycloak as identity provider, while the authentication for the other two services is managed through the ingress controller and [oauth2-proxy](https://github.com/oauth2-proxy/oauth2-proxy).
+
+*Note: this guide assumes Keycloak to be already deployed and available. Please refer to the [keycloak deployment guide](../../identity-provider/README.md) for more information.*
+
+#### Keycloak configuration
+
+1. Create a new client for `monitoring`;
+2. Configure the Client Protocol to be `openid-connect`;
+3. Set Access Type to `confidential`;
+4. Configure the Valid Redirect URIs to the alertmanager, grafana and prometheus URLs (e.g. https://alertmanager.example.com/*, https://grafana.example.com/*, https://prometheus.example.com/*); <!-- markdown-link-check-disable-line -->
+5. From the Credentials tab, copy the Client Secret that has been generated;
+6. From the Mappers tab, add a new `Group Membership` mapper with Token Claim Name equal to `groups`.
+
+#### Alertmanager and Prometheus configuration
+
+Edit the ingress in the [kube-prometheus-stack file values](./kube-prometheus-stack-values.yaml) and adapt it to your configuration. In particular, it is necessary to instert the `annotations`, specify the `hosts`, and configure the `tls`. This needs to be done for [alertmanager ingress](./kube-prometheus-stack-values.yaml#L268) and [prometheus ingress](./kube-prometheus-stack-values.yaml#L2067)
+
+#### Grafana configuration
+
+Edit the [grafana.ini config](./kube-prometheus-stack-values.yaml#L783) and adapt it to your configuration (each line corresponds to an environment variable). In particular, it is necessary to adapt the different URIs and specify the Client ID and Secrets generated in Keycloak. The meaning of the different fields is specified by the embedded comments. More information can be found in the [official documentation](https://grafana.com/docs/grafana/latest/auth/generic-oauth/).
+
+#### Limit access to a subset of Keycloak users
+
+The access to Alertmanager and Prometheus is limited to users belonging to the `monitoring` group. Hence, it is possible to grant access to the users by simply adding them to the `monitoring` group.
+
+**Warning:** At the time of writing, in Grafana it seems not possible to restrict login by role/group when using the `generic_oauth` feature. Hence, all valid users of the Keycloak realm would be able to access Grafana. As a temporary workaround, the [grafana.ini](./kube-prometheus-stack-values.yaml#L782) disables the access to users not already present within the Grafana database. As soon as this [PR](https://github.com/grafana/grafana/pull/22383) is merged and the new version of Grafana released, it should be possible to limit the access to only a subset of Keycloak users directly from grafana, without needing to create duplicated accounts.
```

Comment:
```suggestion
**Warning:** At the time of writing, in Grafana it seems not possible to restrict login by role/group when using the `generic_oauth` feature. Hence, all valid users of the Keycloak realm will be able to access Grafana.
```

The following is deleted, since it does no longer correspond to our configuration.